src/auth/negotiate/kerberos/negotiate_kerberos_pac.cc

```diff
@@ -49,6 +49,11 @@ static unsigned char *p;
 extern int
 check_k5_err(krb5_context context, const char *function, krb5_error_code code);
 
+static char *
+get_resource_group_domain_sid(uint32_t ResourceGroupDomainSid);
+static char *
+get_resource_groups(char *ad_groups, char *resource_group_domain_sid, uint32_t ResourceGroupIds, uint32_t ResourceGroupCount);
+
```

Comment:
Previously, I did not include the definitions in the code, but it did not pass github automatic checks. 
So I added these lines to negotiate_kerberos.h and moved them to negotiate_kerberos_pac.cc after the Alex's review.

Reply:
@ankor2023, I suspect that you have misinterpreted the meaning of an earlier build [error](https://github.com/squid-cache/squid/actions/runs/6952672981/job/18934998337#step:4:5009) for the original PR commit 8450f61ff4fb2a5015ead043b11a31d744262618:

    src/auth/negotiate/kerberos/negotiate_kerberos_pac.cc:411:1: error: no previous declaration for
    ‘char* get_resource_groups(char*, char*, uint32_t, uint32_t)’ [-Werror=missing-declarations]


In that commit, get_resource_groups() was a "global" function -- something other translation units can call directly by name. The compiler was telling us that a _global_  function needs a declaration. 

```
missing-declarations
  Warn if a global function is defined without a previous
  declaration.  Do so even if the definition itself provides a
  prototype.  Use this option to detect global functions that are not
  declared in header files.
```

Today, get_resource_groups() is a local function. When a local static function is not used prior to its definition, it does not need a declaration. Naturally, Squid code is not consistent with avoiding needless local function declarations, but it is best to avoid them.

Reply:
@rousskov : Alex, thank you for the clarifications!